web/packages/extension/js/popup.js

```diff
@@ -34,8 +34,13 @@ function bind_boolean_setting(checkbox_elem) {
     get_obj[name] = default_val;
 
     get_sync_storage(get_obj, function (items) {
-        checkbox_elem.checked = items[name] === checkbox_elem.value;
-        settings_dict[name] = items[name];
+        if ((name == "ruffle_enable") && (items[name] === undefined)) {
+            checkbox_elem.checked = true;
+            settings_dict[name] = "on";
+        } else {
+            checkbox_elem.checked = items[name] === checkbox_elem.value;
+            settings_dict[name] = items[name];
+        }
         on_settings_change_intent();
     });
```

Comment:
The issue is that "ruffle_enable" returns `true` the first time it appears instead of `"on"`, due to the `default_val = checkbox_elem.checked;` on popup.js line 29.

 `true` causes checkbox to go to false on line 41 because of `items[name] === checkbox_elem.value` which does `true === "on"`, returning false.

Some thoughts:
 * Is there a reason we are expect to store "on" and ""? I think it'd just be better as true/false.
 * The "ruffle_enable" special case feels awkward here, and grabbing the default value from the checkbox state seems sketchy as well (we'd have to keep both popup.html and settings.html in sync). I think it'd be best if we had an explicit object with the default values somewhere, and we can grab the default values from that (maybe pass the default_val in as a parameter to this function).

Reply:
I guess a quick fix is to do `default_val = checkbox_elem.checked ? "on" : ""` on line 29.